### PR TITLE
Only do a single push, pushing tags at the same time

### PR DESCRIPTION
**Pull request by aslakhellesoy**
### 🤔 What's changed?

The tag is pushed at the same time as the release branch

### ⚡️ What's your motivation? 

We're using "trigger by tag push" in [cucumber/vscode](https://github.com/cucumber/vscode/blob/main/.github/workflows/release-github.yml) because we had to use a 3rd party Action to be able to upload release assets, and this action expects to find the git tag in ${{ github.ref_name }}

Prior to this change the GHA job running for the release change would not have access to the tag name

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)

### ♻️ Anything particular you want feedback on?

Does it make sense?

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Review by mpkorstanje — CHANGES_REQUESTED** [review]
We've got tests. You gotta fix them. :smile:

File: polyglot-release [hunk 1]
```diff
@@ -400,9 +400,7 @@ local_branch=$(git rev-parse --abbrev-ref HEAD)
 release_commit=$(git rev-list --max-count=1 "$TAG")
 release_branch="release/$TAG"
 if [[ -z $NO_GIT_PUSH ]]; then
-  git push --quiet origin "refs/heads/$local_branch" "refs/tags/$TAG"
-  log "All commit(s) pushed to origin/$local_branch"
-  git push --quiet origin "$release_commit:refs/heads/$release_branch"
+  git push --quiet --tags origin "$release_commit:refs/heads/$release_branch"
```
Comment (mpkorstanje) [inline polyglot-release:403]: This pushes more then just the intended tag. Any reachable tag will be used. 

It also doesn't push the current branch (i.e. `main`) to the `origin/main`.

**Comment by mattwynne** [comment]
I think we can just push all three refs (`refs/heads/main`, `refs/tags/vX.Y.Z` and `refs/heads/release/vX.Y.Z` in a single command, no?

```
$refs = "refs/heads/$local_branch refs/tags/$TAG $release_commit:refs/heads/$release_branch"
git push --quiet origin $refs
```

Would that work as you've intended? @mpkorstanje can you see any reason not to do all this in a single push (assuming git understands it - I haven't tested it)?

**Comment by mpkorstanje** [comment]
Is the failure mode atomic? If one ref fails to push, does nothing get pushed?

**Comment by mattwynne** [comment]
> Is the failure mode atomic? If one ref fails to push, does nothing get pushed?

Looks like it could be, if you use git 2.4: https://github.blog/2015-04-30-git-2-4-atomic-pushes-push-to-deploy-and-more/

**Review by mpkorstanje — APPROVED** [review]

**Comment by mpkorstanje** [comment]
Merged because `--atomic` is better.  But I really do wonder if it fixes the problem because the original already pushed the tags. 

```
git push --quiet origin "refs/heads/$local_branch" "refs/tags/$TAG"
```

**Pull request merged**